### ISSUE TITLE
[P2][dogfood] Kotlin: local val/var inside function bodies are phantom-extracted as class fields

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#759]
**Severity:** P2  |  **Surface:** CLI --advanced --table json (Kotlin extraction, fields[])
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
fields[] includes line 63 (`val manager = UserManager(null)`), a local val inside fun main(); field_count=4. Minimal probe confirms `val local`/`var localVar` inside a function body are also extracted as fields. Local declarations are walked as if they were class properties. All extracted fields also have name='unknown' (name resolution failing too, but the core defect is phantom inclusion of locals).

### Expected
Local variable declarations inside a function body must NOT be reported as fields/properties. Only class/object/top-level member properties belong in fields[]. field_count should be 3 for Sample.kt (userCount line 30, MAX_USERS line 48, version line 49), and field_count should be 1 for the probe (only `val real`).

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood && uv run python -m tree_sitter_analyzer examples/Sample.kt --advanced --table json 2>/dev/null | python3 -c "import json,sys; d=json.load(sys.stdin); print([f['line_range'] for f in d['fields']])"
# -> [{'start': 30, 'end': 31}, {'start': 48, 'end': 48}, {'start': 49, 'end': 49}, {'start': 63, 'end': 63}]
# line 63 is `val manager = UserManager(null)` inside fun main(args)
# Minimal probe (file inside project dir): class Foo { val real=1; fun topLevel(){ val local=1; var localVar=2 } } -> field_count=3 (lines 2,4,5); locals at 4 and 5 are phantom.
```

### Verifier notes
Reproduced the exact repro on develop (HEAD 471ac5f5): fields[] for examples/Sample.kt contains an entry at line_range {start:63,end:63}, which is `val manager = UserManager(null)` — a LOCAL val inside the body of `fun main(args)`. field_count=4 (the 3 legitimate properties at lines 30 userCount, 48 MAX_USERS, 49 version, plus the phantom at 63). A fresh minimal probe (class Foo with one real `val real` and a method containing `val local`/`var localVar`) returns field_count=3: both function-body locals are extracted as class fields. This is a genuine extractor defect (Kotlin extractor walks property_declaration / variable declarations regardless of whether they are class-level members or local declarations inside a function body), not a flag/user error — `--advanced --table json` is the documented extraction path and fields[] is meant for class/object/top-level properties only. Dedup: searched open issues for Kotlin / phantom field / local variable and reviewed round-1 issues #734-#757; none cover Kotlin local-var phantom fields. #747 is JS-specific destructuring with a different mechanism; no overlap.

**Comment by aimasteracc** [comment]
Fixed in PR #834 (commit b80ac0bc) —  now returns None for block-parented nodes (local vars). Closing.
